src/Database/Redis/Connection.hs

```diff
@@ -249,13 +250,24 @@ connectCluster bootstrapConnInfo = do
       clusterConnect :: Bool -> IO Cluster.Connection -> IO Cluster.Connection
       clusterConnect readOnlyConnection connection = do
           clusterConn@(Cluster.Connection nodeMap _ _ _ _) <- connection
-          nodesConns <-  sequence $ ( PP.fromCtx . (\(Cluster.NodeConnection ctx _ _) -> ctx ) . snd) <$> (HM.toList nodeMap)
+          nodesConns <-  sequence $ (ctxToConn . snd) <$> (HM.toList nodeMap)
           when readOnlyConnection $
-                  mapM_ (\conn -> do
-                          PP.beginReceiving conn
-                          runRedisInternal conn readOnly
+                  mapM_ (\maybeConn -> case maybeConn of
+                            Just conn -> do
+                                PP.beginReceiving conn
+                                runRedisInternal conn readOnly
+                            Nothing -> return $ Right (T.Status "Connection does not exist")
                       ) nodesConns
           return clusterConn
+          where
+          ctxToConn :: Cluster.NodeConnection -> IO (Maybe PP.Connection)
+          ctxToConn (Cluster.NodeConnection pool _ nid) = do
+            maybeConn <- try $ withResource pool PP.fromCtx
```

Comment:
Do we have to remove the connection from pool if there is an exception? rather than catching with `try` and returning `Nothing`. Verify this once.